src/Compilers/CSharp/Portable/BoundTree/BoundCollectionExpression.cs

```diff
@@ -2,17 +2,26 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Linq;
-
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class BoundCollectionExpression
+    internal partial class BoundCollectionExpressionBase
     {
-        internal int? GetKnownLength()
+        internal bool HasSpreadElements(out int numberIncludingLastSpread, out bool hasKnownLength)
```

Comment:
I think a doc comment with an example of what `IncludingLastSpread` looks like would be helpful for future maintainability here.